query_builders/query_builder.py

```diff
@@ -328,7 +361,18 @@ def check_connectedness(labels: list, rels: list):
             return False
 
     @staticmethod
-    def enrich_labels_from_rels(labels: list, rels: list, oclass_marker: str):
+    def enrich_labels_from_rels(labels: list, rels: list, oclass_marker: str) -> list:
+        """
+        Add any labels found in relationships that are not already present in labels.
+        As sometimes a label is not optional when a relationship it is involved in IS optional, 
+        we decide on whether to make that label optional.
```

Comment:
I found this wording a little bit confusing. Perhaps it should be something along the lines of:

Extends the labels list with any labels that are only present in relationships. Conditionally non optional labels may become optional during this process if they are involved in an relationship that is optional.

Reply:
Sounds much better, thanks!